Add AM3 specific run docs (#1058)
ACCESS-Hive/access-hive.github.io: docs/models/run_a_model/run_access-am3.md

Author: Well we would have similar information for both- simply links to other locations, myNCI to request to join NCI projects, and to the forum to request access to our repos. I can only think of 3 possible avenues for people to request access:

* The forum, either by replying to a thread or directly messaging one of the listed people.
* Github, via an issue on the access-am3-configs-doc repo
* Direct email to one of the repo admins

I don't think the last option is a good one, and the first 2 both have downsides. For the forum, it has the extra step of requiring people to make an account (which we would encourage people to do anyway). For Github, I think it would be easier for a reply on an issue to get lost in the noise, and there are no direct messaging options AFAIK.

Did you have something else in mind, other than a link to either a publicly available Github issue or a forum topic?

{% set model = "ACCESS-AM3" %}
{% set github_configs = "https://github.com/ACCESS-NRI/access-am3-configs" %}
{% set github_ssh = "git@github.com:ACCESS-NRI/access-am3-configs.git" %}
{% set configs_docs = "https://access-am3-configs.access-hive.org.au" %}
{% set example_branch = "main" %}

> {% set release_notes = "" %}

Reviewer: ```suggestion
{% set release_notes = "https://forum.access-hive.org.au/t/access-am3-release-information/5446" %}
```

# Run {{ model }}

## About

The instructions below outline how to run {{ model }} using ACCESS-NRI's software deployment pipeline, specifically designed to run on the [National Computational Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_](https://nci.org.au/our-systems/hpc-systems).

If you are unsure whether {{ model }} is the right choice for your experiment, take a look at the overview of [ACCESS Models](/models).

> All {{model}} configurations are licensed under the UKMO's [Momentum licence](). {{model}} is delivered to the community through private GitHub repositories. See the [Prerequisites](#prerequisites) section for details.

Author: We decided to just not link to anything when we talk about the Momentum license for now.

        {: style="list-style-type: disc"}

    !!! warning
        You will not be granted access to some of the projects listed here before we have checked you are properly licensed to use the software. Ensure you request access to the configurations first.

    For more information on joining specific NCI projects, refer to [How to connect to a project](https://opus.nci.org.au/display/Help/How+to+connect+to+a+project).

## Get {{model}} configuration

Follow the instructions in the [_Rose/Cylc_ page](/models/run-a-model/rose-cylc.md) using the following specific information in the [Models configurations stored on _GitHub_](/models/run-a-model/rose-cylc/#model-configurations-stored-on-github) section:

- [Repository](/models/run-a-model/rose-cylc/#model-configurations-stored-on-github): {{ github_ssh }}
- [Branch](/models/run-a-model/rose-cylc/#model-configurations-stored-on-github): {{ example_branch }}

## Setup the configuration

Before you can run the configuration, you need to specify which projects you want to use for data storage and compute costs. For this, in the configuration you have just retrieved from GitHub, open the `rose-suite.conf_nci_gadi` file and change:

> - `root_dir` to the path you want to use as a work directory for running the simulation. A space under /scratch is ideal. The directory will be created by the suite if it does not exist.

Author: There's not really any such thing as "your `/scratch` directory`". Perhaps a rewording like:

"A space under `/scratch/<project>` is ideal, where `<project>` is the project associated with the current work."

- [Repository](/models/run-a-model/rose-cylc/#model-configurations-stored-on-github): {{ github_ssh }}
- [Branch](/models/run-a-model/rose-cylc/#model-configurations-stored-on-github): {{ example_branch }}

## Initial Setup

Before you can run the configuration, you need to specify which projects you want to use for data storage and compute costs. For this, in the configuration you have just retrieved from GitHub, open the `rose-suite.conf_nci_gadi` file and change:

- `root_dir` to the path you want to use as a work directory for running the simulation. A space under `/scratch/<project>` is ideal, where `<project>` is the project associated with the current work. The directory will be created by the suite if it does not exist.
- `STORAGE_PROJECT` must be the same project as used in the `root_dir` path.
- `COMPUTE_PROJECT` to any project you want to use for the compute cost.

Now the configuration can be run using the [Run the model configuration](/models/run-a-model/rose-cylc/#run-the-model-configuration) instructions.

## Inspecting the outputs

> The outputs are placed in the directory specified in `STORAGE_PROJECT`, under `cylc-run/<suite-id>`.

Reviewer: ```suggestion
## Inspecting the outputs

The netCDF outputs are placed in `~/cylc-run/<suite-id>/share/data/History_Data/netCDF`.
The outputs in UMFields format are placed in `~/cylc-run/<suite-id>/share/data/History_Data`.
```

Would that be more useful info?

{% set model = "ACCESS-AM3" %}
{% set github_configs = "https://github.com/ACCESS-NRI/access-am3-configs" %}
{% set github_ssh = "git@github.com:ACCESS-NRI/access-am3-configs.git" %}
{% set configs_docs = "https://access-am3-configs.access-hive.org.au" %}
{% set example_branch = "main" %}
{% set release_notes = "https://forum.access-hive.org.au/t/access-am3-release-information/5446" %}

# Run {{ model }}

## About

The instructions below outline how to run {{ model }} using ACCESS-NRI's software deployment pipeline, specifically designed to run on the [National Computational Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_](https://nci.org.au/our-systems/hpc-systems).

If you are unsure whether {{ model }} is the right choice for your experiment, take a look at the overview of [ACCESS Models](/models).

> All {{model}} configurations are licensed under the UKMO's Momentum licence. {{model}} is delivered to the community through private GitHub repositories. See the [Prerequisites](#prerequisites) section for details.

Reviewer 2: ```suggestion
All {{model}} configurations are licensed under the UKMO's Momentum licence. {{model}} is delivered to the community through private GitHub repositories. See the [Request access prerequisite](#request-access) for details.
```

{{ model }} release notes are [available on the ACCESS-Hive Forum]({{release_notes}}) and are updated when new releases are made available.

## Prerequisites

!!! warning

>     If you have not yet read the guide on using the _Rose/Cylc_ tool, please read our [_Rose/Cylc_ documentation](/models/run-a-model/rose-cylc.md) before continuing.

Reviewer 2: ```suggestion
!!! warning
    Before continuing, make sure you have read the guide on [running models using _Rose/Cylc_](/models/run-a-model/rose-cylc.md).
```

In addition to the [prerequisites for _Rose/Cylc_](/models/run-a-model/rose-cylc/#prerequisites), you will need:

- **Request access to the configurations**<br>
    To request access to the configurations repository (and associated model component repositories), please contact us [through the ACCESS-Hive Forum](https://forum.access-hive.org.au/t/request-access-to-am3-configurations/5580/13). This step will also allow us to check you are properly licensed to use the software.

- **Join NCI projects**<br>
    Join the following projects by requesting membership on their respective NCI project pages:

    - [access](https://my.nci.org.au/mancini/project/access/join)
    - [ki32](https://my.nci.org.au/mancini/project/ki32/join)
    - [ki32_mosrs](https://my.nci.org.au/mancini/project/ki32_mosrs/join)
    - [vk83](https://my.nci.org.au/mancini/project/vk83/join)
    - [xp65](https://my.nci.org.au/mancini/project/xp65/join)

    !!! tip
        To request membership for the _ki32_mosrs_ subproject, you need to:
        
        - already be member of the _ki32_ project

>         {: style="list-style-type: disc"}

Reviewer 2: This works, but in this case I think we always want first-level dot points within an admonition to be rendered this way, so it might be preferable to set a general rule in the CSS file.
I suggest removing this line:
```suggestion
```
 and changing the `access-nri.css`
https://github.com/ACCESS-NRI/ACCESS-Hive-Docs/blob/9e8d910e1a1aef83a78745d4083cce7d1e6df7b6/docs/css/access-nri.css#L460-L466

to
```diff
/* =============================================================== 
  Admonitions 
  (https://squidfunk.github.io/mkdocs-material/reference/admonitions/)
+*/
+
+  .admonition > ul li {
+   list-style-type: disc !important;
+ }

/* General style admonition (In markdown --> `!!! ...`) */
.admonition {
```

>         - have a [MOSRS account](#mosrs-account)

Reviewer 2: ```suggestion
        - have a [MOSRS account](/models/run_a_model/rose_cylc/#mosrs-account)
```

>         {: style="list-style-type: disc"}

Reviewer 2: ```suggestion
```
Same as above. With the CSS addition, this is not needed.

    !!! warning
        You will not be granted access to some of the projects listed here before we have checked you are properly licensed to use the software. Ensure you request access to the configurations first.

    For more information on joining specific NCI projects, refer to [How to connect to a project](https://opus.nci.org.au/display/Help/How+to+connect+to+a+project).

## Get {{model}} configuration

> Follow the instructions in the [_Rose/Cylc_ page](/models/run-a-model/rose-cylc.md) using the following specific information in the [Model configurations stored on _GitHub_](/models/run-a-model/rose-cylc/#model-configurations-stored-on-github) section:

Reviewer 2: ```suggestion
Follow the instructions for [Model configurations stored on _GitHub_](/models/run_a_model/rose_cylc/#model-configurations-stored-on-github) using the following specific information:
```

- [Repository](/models/run-a-model/rose-cylc/#model-configurations-stored-on-github): {{ github_ssh }}

> - [Branch](/models/run-a-model/rose-cylc/#model-configurations-stored-on-github): {{ example_branch }}

Reviewer 2: Hmm right. Ok for now this is fine then. 
Probably we will need a section on setting up SSH keys for GitHub then (maybe in the getting started page).

## Initial Setup

Before you can run the configuration, you need to specify which projects you want to use for data storage and compute costs. For this, in the configuration you have just retrieved from GitHub, open the `rose-suite.conf_nci_gadi` file and change:

- `root_dir` to the path you want to use as a work directory for running the simulation. A space under `/scratch/<project>` is ideal, where `<project>` is the project associated with the current work. The directory will be created by the suite if it does not exist.

> - `STORAGE_PROJECT` must be the same project as used in the `root_dir` path.

Reviewer 2: This is not necessarily true.
Otherwise, `STORAGE_PROJECT` could be retrieved automatically from `root_dir`, without the need for the user to change/specify two settings.

- `COMPUTE_PROJECT` to any project you want to use for the compute cost.

Now the configuration can be run using the [Run the model configuration](/models/run-a-model/rose-cylc/#run-the-model-configuration) instructions.

## Inspecting the outputs

The netCDF outputs are placed in `~/cylc-run/<suite-id>/share/data/History_Data/netCDF`.

> The outputs in UMFields format are placed in `~/cylc-run/<suite-id>/share/data/History_Data`.

Reviewer 2: ```suggestion
The outputs in [UM Fieldsfiles format](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf) are placed in `~/cylc-run/<suite-id>/share/data/History_Data`.
```

## Further Information

> See the [ACCESS-AM3 model configuration documentation](https://access-nri.github.io/access-am3-configs-doc/) for more detailed information about the configuration.

Reviewer 2: ```suggestion
See [{{ model }} configuration documentation](https://access-nri.github.io/access-am3-configs-doc/) for more detailed information about {{ model }} configuration.
```
